components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -192,4 +206,28 @@ private OrganizationManager getOrganizationManager() {
 
         return OrganizationSwitchGrantDataHolder.getInstance().getOrganizationManager();
     }
+
+    private OrgApplicationManager getOrgApplicationManager() {
+
+        return OrganizationSwitchGrantDataHolder.getInstance().getOrgApplicationManager();
+    }
+
+    private ApplicationManagementService getApplicationManagementService() {
+
+        return OrganizationSwitchGrantDataHolder.getInstance().getApplicationManagementService();
+    }
+
+    private String getAppID(String appName, String tenantDomain) throws IdentityOAuth2Exception {
+        try {
+            ApplicationBasicInfo applicationBasicInfo = getApplicationManagementService().
+                    getApplicationBasicInfoByName(appName, tenantDomain);
+            if (applicationBasicInfo.getApplicationResourceId() != null) {
+                return applicationBasicInfo.getApplicationResourceId();
+            } else {
+                throw new IdentityOAuth2Exception("Application ID is null for the application: " + appName);
```

Comment:
What is the error msg was like Application not found by the name..

Reply:
Fixed